e2e/app/element/container.ts

```diff
@@ -0,0 +1,19 @@
+export default class Container {
+
+  private readonly _xpath: string;
+
+  constructor(selector: {xpath?: string, testId?: string}) {
+    if (selector.testId !== undefined) {
+      this._xpath = `//*[@data-test-id="${selector.testId}"]`;
+    } else if (selector.xpath !== undefined) {
+      this._xpath = selector.xpath;
+    } else {
+      throw new Error('Require selector xpath or testId.');
+    }
+  }
+
+  get xpath() {
```

Comment:
So `Container` does not extend `BaseElement`, yet defines its own `xpath` accessor. This means we have composition going on at the same time as inheritance, so the tree is also a bush kind of... it's a mixed metaphor, unless I'm misunderstanding.

I think if you're going to go this route, then you have to be really careful, probably write a document about the scheme, and make sure you teach folks how it's intended to be used. For example if a Container is an Element but not a Base Element, but BaseElement is the parent for fall the elements, then is there a contradiction.